vowpalwabbit/vw_exception.h

```diff
@@ -104,6 +111,31 @@ class strict_parse_exception : public vw_exception
   ~strict_parse_exception() noexcept override = default;
 };
 
+inline std::string strerror_to_string(int error_number)
+{
+#ifdef _WIN32
+  constexpr auto BUFFER_SIZE = 256;
+  std::array<char, BUFFER_SIZE> error_message_buffer;
+  auto result = strerror_s(error_message_buffer.data(), error_message_buffer.size(), error_number);
+  if (result != 0) { return "unknown"; }
+
+  auto length = std::strlen(error_message_buffer.data());
+  return std::string(error_message_buffer.data(), length);
+#else
+  // Passing "" for the locale means use the default system locale
+  locale_t locale = newlocale(LC_ALL_MASK, "", static_cast<locale_t>(0));
+
+  if (locale == static_cast<locale_t>(0)) {
+    return "Failed to create locale";
```

Comment:
on failure newlocale sets errorno to specific error cases. Not sure if we want to incorporate that for beefier error messages. Not sure if this path is worthy of it though

Reply:
Yeah, I thought about that. But then to get the error message you need to create a locale - the exact thing that failed... It seems like a sort of fail loop?

Reply:
would it be worth it to report the number only?

Reply:
aren't those lookable or fixed? I actually don't remember

Reply:
Yeah we could just report out the number so the user can look it up. Seems like a reasonable backup